App/Program.fs

```diff
@@ -22,46 +23,50 @@ let writeUrl =
 let readUrl =
   Environment.tryGetEnv "MESSAGE_DB_REPLICA_URL" |> Option.defaultValue writeUrl
 
-let connection = MessageDbConnector(writeUrl, readUrl).Establish()
-let context = MessageDbContext(connection)
+let connection = MessageDbClient(writeUrl, readUrl)
+let context = MessageDbContext(connection, 100)
 let caching = CachingStrategy.SlidingWindow(cache, TimeSpan.FromMinutes(20))
 
 let service =
   MessageDbCategory(context, Invoice.Events.codec, Invoice.Fold.fold, Invoice.Fold.initial, caching)
   |> Equinox.Decider.resolve log
   |> Invoice.create
```

Comment:
Maybe shift this into an `InvoiceService.create (context, cache)` that encapsulates creating the CachingStrategy, MessageDbCategory and the Decider.resolve ? (Same for numbering service)

(I know it seems clear and terse here, but the fact you are connecting 2 services to a single context will stand out better)